engine/src/test/java/io/camunda/zeebe/engine/state/distribution/DistributionStateTest.java

```diff
@@ -56,15 +57,16 @@ public void shouldAddPendingDistribution() {
   @Test
   public void shouldRemovePendingDistribution() {
     // given
-    final var deploymentKey = 10L;
+    final var distributionKey = 10L;
     final var partition = 1;
-    distributionState.addPendingDistribution(deploymentKey, partition);
+    distributionState.addCommandDistribution(distributionKey, createCommandDistributionRecord());
+    distributionState.addPendingDistribution(distributionKey, partition);
 
     // when
-    distributionState.removePendingDistribution(deploymentKey, partition);
+    distributionState.removePendingDistribution(distributionKey, partition);
 
     // then
-    assertThat(distributionState.hasPendingDistribution(deploymentKey)).isFalse();
+    assertThat(distributionState.hasPendingDistribution(distributionKey)).isFalse();
```

Comment:
💭 In your commit message you describe

>that as long as we have a pending deployment, the ValueType and RecordValue need to remains stored. We can guarantee this by using a foreign key.

I agree with the requirement, but this is not guaranteed by our foreign keys. The foreign keys are only used to check that an entry with that key exists in the foreign column family when:
- we insert a new entry in the referring column family
- we update an entry in the referring column family
- we upsert an entry in the referring column family

That's not to say that we won't add such a check in the future, but we don't support it, at this time.

🔧 It may be a good idea to add a test case that asserts the foreign key usage:
 - add a pending distribution with an unstored command distribution should fail